internal/monitors/telegraf/common/emitter/baseemitter/baseemitter.go

```diff
@@ -137,6 +140,35 @@ func (b *BaseEmitter) FilterTags(key string, value string) (include bool) {
 	return include
 }
 
+// RenameMetric adds a mapping to rename a metric by it's name
+func (b *BaseEmitter) RenameMetric(original string, override string) {
```

Comment:
Unused.  Also does it really need to be thread safe since metric name mapping will only happen on `Configure` and given that monitor instances are no longer reconfigured?  If not, take out the locks, esp the reader lock in `GetMetricName` since that is on a somewhat hot path and the read lock still involves mutating shared memory across threads which will kill performance at high DPM.

Reply:
GetMetricName could be extended and used concurrently.  This is a base struct feature.

That said, you're right that configuration of the map of metrics to exclude will probably never be accessed concurrently.

Reply:
And RenameMetric will be used in a future PR